presto-native-execution/presto_cpp/main/common/Configs.h

```diff
@@ -549,6 +549,13 @@ class NodeConfig : public ConfigBase {
 
   uint64_t nodeMemoryGb(
       const std::function<uint64_t()>& defaultNodeMemoryGb = nullptr) const;
+
+  /// Invoked to get the ip address of the process. In certain deployment
+  /// setup, each process has different ip address. Deployment environment
+  /// may provide there own library to get process specific ip address.
+  /// In such cases, getLocalIp can be overriden to pass process specific
+  /// ip address.
+  virtual std::string getLocalIp() const;
```

Comment:
Its reasonable to have this function in PrestoServer itself because its virtual and we are allowing derived PrestoServer classes (Meta and others have it to plug-in their custom metrics collection, error reporting etc) to override it.

With this change services will have to override NodeConfig class.